src/game/UICmdList.cpp

```diff
@@ -0,0 +1,283 @@
+    __ASSERT(m_pList_Cmds, "NULL UI Component!!!");
+
+    //CN3UIArea
+    // CN3UIBase * pChild = m_pArea_Destroy->GetChildByIndex(i);
+    /*
+    m_pScroll_Categories = (CN3UIList *)m_pList_CmdCat->GetChildByIndex(0);
+    __ASSERT(m_pScroll_Categories, "NULL UI Component!!");
+    m_pScroll_Categories->SetVisible(true);
+
+    m_pScroll_Commands = (CN3UIScrollBar *)m_pList_Cmds->GetChildByIndex(0);
+    __ASSERT(m_pScroll_Categories, "NULL UI Component!!");
+    m_pScroll_Commands->SetVisible(true);
+
+    */
+
+    CreateCategoryList();
+
+    return true;
+}
+
+void CUICmdListDlg::Release() {
+    CN3UIBase::Release();
+}
+
+void CUICmdListDlg::Open() {
+    SetVisible(true);
+    this->SetPos(CN3Base::s_CameraData.vp.Width, 10);
+    m_fMoveDelta = 0;
+    m_bOpenningNow = true;
+    m_bClosingNow = false;
+
+    m_iRBtnDownOffs = -1;
+}
+void CUICmdListDlg::Close(bool bByKey) {
+    SetVisible(false);
+    m_fMoveDelta = 0;
+    m_bOpenningNow = false;
+    m_bClosingNow = true;
+}
+
+void CUICmdListDlg::Tick() {
+    if (m_bOpenningNow) {
+        POINT ptCur = this->GetPos();
+        RECT  rc = this->GetRegion();
+        float fWidth = (float)(rc.right - rc.left);
+
+        float fDelta = 5000.0f * CN3Base::s_fSecPerFrm;
+        fDelta *= (fWidth - m_fMoveDelta) / fWidth;
+        if (fDelta < 2.0f) {
+            fDelta = 2.0f;
+        }
+        m_fMoveDelta += fDelta;
+
+        int iXLimit = CN3Base::s_CameraData.vp.Width - (int)fWidth;
+        ptCur.x = CN3Base::s_CameraData.vp.Width - (int)m_fMoveDelta;
+        if (ptCur.x <= iXLimit) {
+            ptCur.x = iXLimit;
+            m_bOpenningNow = false;
+        }
+
+        this->SetPos(ptCur.x, ptCur.y);
+    } else if (m_bClosingNow) {
+        POINT ptCur = this->GetPos();
+        RECT  rc = this->GetRegion();
+        float fWidth = (float)(rc.right - rc.left);
+
+        float fDelta = 5000.0f * CN3Base::s_fSecPerFrm;
+        fDelta *= (fWidth - m_fMoveDelta) / fWidth;
+        if (fDelta < 2.0f) {
+            fDelta = 2.0f;
+        }
+        m_fMoveDelta += fDelta;
+
+        int iXLimit = CN3Base::s_CameraData.vp.Width;
+        ptCur.x = CN3Base::s_CameraData.vp.Width - (int)(fWidth - m_fMoveDelta);
+        if (ptCur.x >= iXLimit) {
+            ptCur.x = iXLimit;
+            m_bClosingNow = false;
+
+            this->SetVisibleWithNoSound(false, false, true);
+        }
+
+        this->SetPos(ptCur.x, ptCur.y);
+    }
+
+    CN3UIBase::Tick();
+}
+
+bool CUICmdListDlg::CreateCategoryList() {
+    if (m_pList_CmdCat == NULL || m_pList_Cmds == NULL) {
+        return false;
+    }
+
+    std::string szCategory;
+    int         idStart = IDS_PRIVATE_CMD_CAT;
+
+    for (int i = 0; i < 8; i++) {
+        ::_LoadStringFromResource(i + 7800, szCategory); //load command categories
+
+        if (CGameBase::s_pPlayer->m_InfoBase.iAuthority == 0 && szCategory == "GM") { // if not gm hide this category
```

Comment:
This is another case of relying on the content of the TBLs. Ideally it should be using an enum or some other language-agnostic indicator, that isn't explicitly relying on the contents of the TBL.